Two tests are primarily failing.

First:
```
 p, q, r = [set([tuple(i.evalf(2) for i in j) for j in R]) for R in A, B, C]
                                                                        ^
SyntaxError: invalid syntax
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
Exception: Tests failed
```
Shows invalid syntax for python versions >= 3.2

Second:
```
_____________________ sympy.diffgeom.diffgeom.CoordSystem ______________________
File "/home/travis/virtualenv/pypy-2.4.0/site-packages/sympy-0.7.6_git-py2.7.egg/sympy/diffgeom/diffgeom.py", line 118, in sympy.diffgeom.diffgeom.CoordSystem
Failed example:
    rect.coord_tuple_transform_to(polar, [1, 1])
Expected:
    Matrix([
    [sqrt(2)],
    [   pi/4]])
Got:
    Matrix([
    [(-2 + sqrt(2))/(-sqrt(2) + 1)],
    [                         pi/4]])
```
Above test fails primarily when `FASTCACHE=False` and once when `TEST_ASCII=True`

First error seems easy enough, but I don't know why exactly second is failing. Why is `FASTCACHE=False` causing problem ?

--- a/sympy/diffgeom/diffgeom.py
+++ b/sympy/diffgeom/diffgeom.py
@@ -95,6 +95,7 @@ class CoordSystem(Basic):
 
     >>> from sympy import symbols, sin, cos, pi
     >>> from sympy.diffgeom import Manifold, Patch, CoordSystem
+    >>> from sympy.simplify import simplify
     >>> r, theta = symbols('r, theta')
     >>> m = Manifold('M', 2)
     >>> patch = Patch('P', m)
@@ -115,7 +116,7 @@ class CoordSystem(Basic):
     Matrix([
     [0],
     [2]])
-    >>> rect.coord_tuple_transform_to(polar, [1, 1])
+    >>> rect.coord_tuple_transform_to(polar, [1, 1]).applyfunc(simplify)
     Matrix([
     [sqrt(2)],
     [   pi/4]])

--- a/sympy/solvers/solvers.py
+++ b/sympy/solvers/solvers.py
@@ -1592,7 +1592,6 @@ def _solve(f, *symbols, **flags):
 
 
 def _solve_system(exprs, symbols, **flags):
-    check = flags.get('check', True)
     if not exprs:
         return []
 
@@ -1601,6 +1600,8 @@ def _solve_system(exprs, symbols, **flags):
     failed = []
     result = False
     manual = flags.get('manual', False)
+    check = flags.get('check', True)
+
     for j, g in enumerate(exprs):
         dens.update(_simple_dens(g, symbols))
         i, d = _invert(g, *symbols)
@@ -1643,6 +1644,7 @@ def _solve_system(exprs, symbols, **flags):
                 # is also handled by checksol
                 if any(checksol(d, result, **flags) for d in dens):
                     result = None
+                check = False
             if failed:
                 if result:
                     solved_syms = list(result.keys())
@@ -1694,25 +1696,6 @@ def _solve_system(exprs, symbols, **flags):
                     #
                     result = [dict(list(zip(solved_syms, r))) for r in result]
 
-            if result:
-                # check & simplify nonlinear solutions
-                # simplify first
-                if flags.get('simplfy', True):
-                    for sol in result:
-                        for k in sol:
-                            sol[k] = simplify(sol[k])
-                flags['simplify'] = False  # don't need to do so in checksol now
-                if check:
-                    ok = []
-                    for sol in result:
-                        if any(checksol(e, sol, **flags) is False for e in exprs):
-                            continue
-                        if any(checksol(d, sol, **flags) for d in dens):
-                            continue
-                        ok.append(sol)
-                    result = ok
-                    del ok
-
     if failed:
         # For each failed equation, see if we can solve for one of the
         # remaining symbols from that equation. If so, we update the
@@ -1734,14 +1717,16 @@ def _solve_system(exprs, symbols, **flags):
 
         solved_syms = set(solved_syms)  # set of symbols we have solved for
         legal = set(symbols)  # what we are interested in
-        simplify_flag = flags.get('simplify', None)
+        missing = Dummy()
+        simplify_flag = flags.get('simplify', missing)
         do_simplify = flags.get('simplify', True)
         # sort so equation with the fewest potential symbols is first
         for eq in ordered(failed, lambda _: len(_ok_syms(_))):
             newresult = []
-            bad_results = []
             got_s = set([])
             u = Dummy()
+            got_s = set([])
+            bad_results = []
             for r in result:
                 # update eq with everything that is known so far
                 eq2 = eq.subs(r)
@@ -1799,8 +1784,7 @@ def _solve_system(exprs, symbols, **flags):
                         # and add this new solution
                         rnew[s] = sol
                         newresult.append(rnew)
-                    if simplify_flag is not None:
-                        flags['simplify'] = simplify_flag
+                    flags['simplify'] = simplify_flag
                     got_s.add(s)
                 if not got_s:
                     raise NotImplementedError('could not solve %s' % eq2)
@@ -1808,6 +1792,25 @@ def _solve_system(exprs, symbols, **flags):
                 result = newresult
             for b in bad_results:
                 result.remove(b)
+            if simplify_flag is missing:
+                flags.pop('simplify', None)
+
+    if check and result:
+        if flags.get('simplfy', True):
+            for sol in result:
+                for k in sol:
+                    sol[k] = simplify(sol[k])
+            flags['simplify'] = False  # don't need to do so in checksol now
+        temp = []
+        for sol in result:
+            if any(checksol(e, sol, **flags) is False for e in exprs):
+                continue
+            if any(checksol(d, sol, **flags) for d in dens):
+                continue
+            temp.append(sol)
+        result = temp
+        del temp
+
     return result
 
 

--- a/sympy/solvers/tests/test_solvers.py
+++ b/sympy/solvers/tests/test_solvers.py
@@ -1025,7 +1025,9 @@ def test_issue_5849():
     Q2: 2*I3 + 2*I5 + 3*I6,
     I1: I2 + I3,
     Q4: -I3/2 + 3*I5/2 - dI4/2}]
-    assert solve(e, I1, I4, Q2, Q4, dI1, dI4, dQ2, dQ4, manual=True) == ans
+    v = I1, I4, Q2, Q4, dI1, dI4, dQ2, dQ4
+    assert solve(e, *v, **dict(manual=True, check=False)) == ans
+    assert solve(e, *v, **dict(manual=True)) == []
     # the matrix solver (tested below) doesn't like this because it produces
     # a zero row in the matrix. Is this related to issue 4551?
     assert [ei.subs(
@@ -1626,3 +1628,33 @@ def test_issue_8755():
     assert len(solve(sqrt(y)*x + x**3 - 1, x)) == 3
     assert len(solve(-512*y**3 + 1344*(x + 2)**(S(1)/3)*y**2 -
         1176*(x + 2)**(S(2)/3)*y - 169*x + 686, y, _unrad=False)) == 3
+
+
+@slow
+def test_issue_8828():
+    x1 = 0
+    y1 = -620
+    r1 = 920
+    x2 = 126
+    y2 = 276
+    x3 = 51
+    y3 = 205
+    r3 = 104
+    v = x, y, z
+
+    f1 = (x - x1)**2 + (y - y1)**2 - (r1 - z)**2
+    f2 = (x2 - x)**2 + (y2 - y)**2 - z**2
+    f3 = (x - x3)**2 + (y - y3)**2 - (r3 - z)**2
+    F = f1,f2,f3
+
+    g1 = sqrt((x - x1)**2 + (y - y1)**2) + z - r1
+    g2 = f2
+    g3 = sqrt((x - x3)**2 + (y - y3)**2) + z - r3
+    G = g1,g2,g3
+
+    A = solve(F, v)
+    B = solve(G, v)
+    C = solve(G, v, manual=True)
+
+    p, q, r = [set([tuple(i.evalf(2) for i in j) for j in R]) for R in [A, B, C]]
+    assert p == q == r

